fe/fe-core/src/main/java/org/apache/doris/datasource/property/storage/OBSProperties.java

```diff
@@ -0,0 +1,88 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.datasource.property.storage;
+
+import org.apache.doris.datasource.property.ConnectorProperty;
+
+import com.google.common.base.Strings;
+import lombok.Setter;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class OBSProperties extends AbstractObjectStorageProperties {
+
+    @Setter
+    @ConnectorProperty(names = {"obs.endpoint", "endpoint", "s3.endpoint"}, required = false,
+            description = "The endpoint of OBS.")
+    protected String obsEndpoint = "";
+
+    @ConnectorProperty(names = {"obs.access_key"}, description = "The access key of OBS.")
+    protected String obsAccessKey = "";
+
+    @ConnectorProperty(names = {"obs.secret_key"}, description = "The secret key of OBS.")
+    protected String obsSecretKey = "";
+
+
+    @ConnectorProperty(names = {"obs.region", "region", "s3.region"}, required = false,
+            description = "The region of OBS.")
+    protected String region;
+
+    public OBSProperties(Map<String, String> origProps) {
+        super(Type.OBS, origProps);
+        // Initialize fields from origProps
+    }
+
+    protected static boolean guessIsMe(Map<String, String> origProps) {
+        return origProps.containsKey("obs.access_key");
+    }
+
+    @Override
+    public void toNativeS3Configuration(Map<String, String> config) {
+        config.putAll(generateAWSS3Properties(obsEndpoint, getRegion(), obsAccessKey, obsSecretKey));
+    }
+
+    public String getRegion() {
```

Comment:
Take care of these `tiny` methods.
These method may be called very frequently in high concurrency scenarios.
But here are 2 issues:
1. You Pattern.compile every time, which is time-consuming.
2. Why not set the `region` when creating this `OBSProperties`, for only once?

Reply:
And I saw the different logic between this and `getRegion` in `OSSProperties`.
In `OSSProperties`, this `region` is set, it will be returned directly without parsing again.